### ISSUE TITLE
Feat: add possibility to expand (drill down) several blocks at a time

### ISSUE THREAD

**Issue by beliaev-maksim** [structurizr/java#358]
### Description

Hello,
We are looking for a diagramming tool that can dynamically render the C4 model. For example, if two systems are defined, each with two containers, we want to define the model once and automatically visualize the connections between different levels and combinations.

I better illustrate this on the example of a simple org chart.

<details><summary>DSL Code</summary>

```dsl
workspace "Name" "Description"

    !identifiers hierarchical

    model {
        department-1 = softwareSystem "department-1" {
            product-managers = container "Product Management"
            market = container "Market Analytics"
        }
        
        department-2 = softwareSystem "department-2" {
            design = container "Design Team"
            branding = container "Branding Team"
        }
        
        department-1.product-managers -> department-2.design "Provides specifications to"
        department-2.design -> department-2.branding "Works with"
    }
    views {
        systemContext department-1 "Diagram1" {
            include *
            autolayout lr
        }

        container department-1 "Diagram2" {
            include *
            autolayout lr
        }
        
        container department-2 "Diagram3" {
            include *
            autolayout lr
        }
    }
}
```
</details> 

which renders perfectly fine into system context:
![image](https://github.com/user-attachments/assets/70a7d300-9f69-4219-9c6c-a794516b8663)

which renders fine from container in department-1 to department-2:
![image](https://github.com/user-attachments/assets/9790b8f7-c617-4f2f-bcdc-6087843864f1)

which renders fine from department-1 to container in department-2:
![image](https://github.com/user-attachments/assets/1bc48992-a58a-400f-885c-732737513033)

however, the difficulty appears if a users wants to drill down into department-1 while looks at the Diagram3 (from department-1 to container in department-2)
Now it is not possible, however, desired would be to expand the block, see PlantUML example:
![image](https://github.com/user-attachments/assets/2c90027b-e3f5-413c-b671-b12b43cf036b)

<details><summary>PlantUML Code</summary>
<p>

```plantuml
@startuml
set separator none
title department-1 - Containers

left to right direction

!include <C4/C4>
!include <C4/C4_Context>
!include <C4/C4_Container>


System_Boundary("department2_boundary", "department-2", $tags="") {
  Container(department2.DesignTeam, "Design Team", $techn="", $descr="", $tags="", $link="")
  Container(department2.BrandingTeam, "Branding Team", $techn="", $descr="", $tags="", $link="")
}

System_Boundary("department1_boundary", "department-1", $tags="") {
  Container(department1.ProductManagement, "Product Management", $techn="", $descr="", $tags="", $link="")
  Container(department1.MarketAnalytics, "Market Analytics", $techn="", $descr="", $tags="", $link="")
}

Rel(department1.ProductManagement, department2.DesignTeam, "Provides specifications to", $techn="", $tags="", $link="")
Rel(department2.DesignTeam, department2.BrandingTeam, "Works with", $techn="", $tags="", $link="")


SHOW_LEGEND(true)
@enduml
```

</p>
</details> 

### Priority

I'm willing to pay for this feature (add details below)

### More information

We can consider paying for the feature or trying to contribute it by ourselves

**Comment by simonbrowndotje** [comment]
Unless I've misunderstood, you can do this already; see https://docs.structurizr.com/dsl/cookbook/container-view-multiple-software-systems/

Example:

```
container department-1 "Diagram4" {
    include "element.parent==department-1"
    include "element.parent==department-2"
    autolayout lr
}
```

![image](https://github.com/user-attachments/assets/cd97cd50-d6d4-4a8b-85f8-708c75410adb)

**Comment by beliaev-maksim** [comment]
> It’s possible to add an explicit view to the DSL example above, but this doesn’t allow for automatic, selectable expansion of blocks. This becomes even more complicated as the number of systems increases, then one would need to generate a matrix of the possible view combinations.
> 
> ```asl
>         container department-1 "AllContainersView" {
>             include department-1.product-managers department-2.design department-2.branding department-1.market
>             autoLayout lr
>         }
> ```

as I wrote in the description that will not allow me the automatic click and drill down. And then I have to specify a matrix of all the possible variations. Or am I missing something ?

**Comment by simonbrowndotje** [comment]
> that will not allow me the automatic click and drill down

Correct, and this is by design. The recommendation is 1 workspace = 1 software system, and I don't recommend showing containers outside of the scoped software system. See my "The C4 Model – Misconceptions, Misuses & Mistakes" talk for the rationale behind this.

If you do want this feature, you'd need to modify the UI -> https://github.com/structurizr/ui

Alternatively, something like Ilograph might provide a better UI.

**Comment by beliaev-maksim** [comment]
thank you for the reply. Let me first get familiar with the talk
